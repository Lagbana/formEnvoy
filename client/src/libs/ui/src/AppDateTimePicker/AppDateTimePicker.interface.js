import { ViewStyle, TouchableOpacityProps } from 'react-native';

export interface AppDateTimePickerProps
  extends Omit<TouchableOpacityProps, 'onChange' | 'onBlur'> {
  value: Date;
  onChange?: (date: Date) => void;
  onBlur?: (visible: boolean, date: Date) => void;
  minimumDate?: Date;
  maximumDate?: Date;
  error?: string;
  style?: ViewStyle;
}
