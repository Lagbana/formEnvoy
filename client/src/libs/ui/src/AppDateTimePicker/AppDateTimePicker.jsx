import React from 'react';
import { Platform } from 'react-native';
import { AppDateTimePicker as AppDateTimePickerNative } from './components/AppDateTimePicker.native';
import { AppDateTimePicker as AppDateTimePickerWeb } from './components/AppDateTimePicker.web';
import { AppDateTimePickerProps } from './AppDateTimePicker.interface';

export const AppDateTimePicker: React.FC<AppDateTimePickerProps> = (props) => {
  switch (Platform.OS) {
    case 'ios':
    case 'android':
      return <AppDateTimePickerNative {...props} />;
    default:
      return <AppDateTimePickerWeb {...props} />;
  }
};
