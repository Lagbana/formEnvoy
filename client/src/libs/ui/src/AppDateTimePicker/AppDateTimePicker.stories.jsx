import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppDateTimePicker } from './AppDateTimePicker'


storiesOf('AppDateTimePicker', module)
  .add('Default', () => (
    <AppDateTimePicker
			value={new Date()}
    />
  ))
	.add('Error', () => (
    <AppDateTimePicker
			value={new Date()}
			onBlur={(visible, date) => { }}
			error={'An error has occurred'}
    />
  ))
	.add('Minimum Date', () => (
    <AppDateTimePicker
			value={new Date()}
			onBlur={(visible, date) => { }}
			minimumDate={new Date()}
    />
  ))
	.add('Maximum Date', () => (
    <AppDateTimePicker
			value={new Date()}
			onBlur={(visible, date) => { }}
			minimumDate={new Date(2020, 11, 24, 10, 33, 30, 0)}
    />
  ))
