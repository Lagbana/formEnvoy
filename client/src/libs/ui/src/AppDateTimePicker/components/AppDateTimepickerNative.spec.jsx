import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppDateTimePicker } from './AppDateTimePicker.native'

configure({ adapter: new Adapter() })

test('<AppDateTimePicker /> renders correctly', () => {
  const component = shallow(<AppDateTimePicker value={new Date('2020-09-23T15:45:00.247Z')} />)
  expect(toJson(component)).toMatchSnapshot()
})
