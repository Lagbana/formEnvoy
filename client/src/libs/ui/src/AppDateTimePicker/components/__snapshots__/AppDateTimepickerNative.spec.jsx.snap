// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppDateTimePicker /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <DateTimePickerModal
    cancelTextIOS="Cancel"
    confirmTextIOS="Confirm"
    date={2020-09-23T15:45:00.247Z}
    display="spinner"
    headerTextIOS="Pick a date"
    isDarkModeEnabled={false}
    isVisible={false}
    modalPropsIOS={Object {}}
    mode="date"
    onCancel={[Function]}
    onConfirm={[Function]}
    onHide={[Function]}
    pickerContainerStyleIOS={Object {}}
  />
  <ForwardRef
    onPress={[Function]}
  >
    <ForwardRef
      editable={false}
      icon="calendar"
      value="September 23, 2020"
    />
  </ForwardRef>
</View>
`;
